### ISSUE TITLE
Custom user roles for Learn WordPress

### ISSUE THREAD

**Issue by varlese** [WordPress/Learn#220]
As part of [this discussion](https://make.wordpress.org/training/2021/05/27/proposal-adding-custom-user-roles-to-learn-wordpress/), I'd like to propose two custom user roles for Learn WordPress. The goal would be to add in these two new user roles to help create more granular user access to better reflect the structure and permissions needed for contributing.

The two new proposed roles would be:
- Workshop Reviewers, which would provide "Deputy"-level access for reviewing workshop applications
- Lesson Plan Editors, which would provide edit access to all Lesson Plans on the site

With this proposal, the full list of user roles on Learn would look like: 

| User Role  | Permissions |
| ------------- | ------------- |
| Author  | Edit access to specific Workshops and specific Lesson Plans |
| Lesson Plan Editors  | Edit access to all Lesson Plans  |
| Editors | Edit access to all Lesson Plans, all Workshops, and Courses | 
| Workshop Reviewers | Edit access to all content types, deputy-level access to meta field for application vetting, and ability to add new users to the site |
| Administrators | Full site access and permissions |

**Comment by iandunn** [comment]
That seems like a good idea, and it's not hard to implement.

WP's role system can be limiting, because users can only have 1 role, but it's probably good enough in this case. On WordCamp.org w/ have a hardcoded list of usernames, and add specific capabilities to them at runtime, but I don't think that's needed in this context. Here we don't need to require proxy access, or grant access to multiple sites.

I'd personally lean towards trusting folks with the regular `Editor` role, rather than creating new ones, but I don't feel strongly. The security issues seem limited, and giving people trust/autonomy reduces friction when someone's contributions take them beyond their normal role. It also has intangible benefits for the social health of teams IMO. The cluttered admin menus could possibly be [solved in a more direct way](https://core.trac.wordpress.org/attachment/ticket/53544/). I don't feel strongly, though, happy to create the new roles too.

**Comment by courtneyr-dev** [comment]
We've bumped into a few issues where folks have editing content that we would rather they not, such as the lesson plan template reusable block. 

Thanks @varlese for summarizing this.

**Comment by varlese** [comment]
That's a helpful example @courane01, thanks!

In practice, I think the _only_ role we're using right now is Editor, since nothing else quite fits the team's workflow. Two places where that really "breaks" are:
- When vetting workshop or course applications, there is some private information included that would benefit from some extra permissions needed (like Deputies and WordCamp/meetup vetting).
- For new contributors, it's pretty common to start contributing through either writing a new lesson plan or reviewing a draft lesson. With that, my understanding is that the Author permissions can get a little clunky and add some more overhead for Training reps when adding new folks to the site to help with those tasks specifically.

On that second point, though, I'd defer to @courane01's thoughts since she works a lot more closely with onboarding new team members!

**Comment by azhiya** [comment]
I'll add my support to the need to have more defined roles for Learn. I'm in the middle of trying to unpick what has happened to 2 lesson plans. The drafts were created (without using the defined style guide) and then subsequently deleted as they went down the workshop route instead. However, someone was willing to work on these as lesson plans and now we have to start from scratch. Therefore a review process before anyone deletes or publishes something in Learn is needed.

**Comment by courtneyr-dev** [comment]
Current user roles:
<img width="233" alt="image" src="https://user-images.githubusercontent.com/5519705/181660753-d76efa3c-43e3-4f5c-8996-70f3e84fe663.png">

- workshop reviewer
- lesson plan editor
- teacher
- subscriber
- contributor
- author
- editor
- administrator

Proposed changes:
- Tutorial reviewer (renaming as the content type has been renamed)
  - Edit access to all content types
  - Deputy-level access to meta field for application vetting
  - Ability to add new users to the site 
 -  Lesson Plan Editors: remove, is redundant with editors
 - Reviewer: edit access to all site, cannot publish own posts without another person reviewing
 - Teacher - it is unclear what this role can do or access - can this be answered?
- Author: Edit access to specific Workshops and specific Lesson Plans
- Editor - Edit access to all Lesson Plans, all Workshops, and Courses, not vetting applicants

**Comment by webtechpooja** [comment]
I agree with Courtney's concern, So many roles might create confusion.  So here we can have a few roles, like:

- **Author** -  Edit access to specific Workshops and specific Lesson Plans
- **Editors** - Edit access to all Lesson Plans, all Workshops, and Courses
- **Tutorial Reviewers** - Edit access to all content types, deputy-level access to meta field for application vetting, and the ability to add new users to the site
- **Administrators** - Full site access and permissions

**Comment by azhiya** [comment]
Agree with both @courtneyr-dev and @webtechpooja. I'd vote for the streamlined version that Pooja has outlined above.
